### PR TITLE
Migrate actions/helpers.js to flow

### PR DESCRIPTION
**Pull request by pepf**
#### Summary
Add flow typing to actions/helpers.js

#### Ticket Link
[MM-11041](https://mattermost.atlassian.net/browse/MM-11041)
https://github.com/mattermost/mattermost-server/issues/8997

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] Ran `make check-style` to check for style errors (required for all pull requests)
- [x] Ran `make test` to ensure unit tests passed
- [x] Ran `make flow` to ensure type checking passed
- [ ] ~Added or updated unit tests (required for all new features)~
- [ ] ~All new/modified APIs include changes to the drivers~

#### Test Information
This PR was tested on: [Mac OS High Sierra, 10.13.5]

### CODE REVIEW

File: src/actions/helpers.js [hunk 1]
```diff
@@ -1,50 +1,58 @@
...
 // See LICENSE.txt for license information.
+// @flow
 
 import {batchActions} from 'redux-batched-actions';
 import {Client4} from 'client';
 import {UserTypes} from 'action_types';
 import {logError} from './errors';
+
+import type {Client4Error} from '../types/errors';
+import type {GenericAction, ActionResult, DispatchFunc, GetStateFunc} from '../types/actions';
```
Comment (jwilander) [inline src/actions/helpers.js]: Can you switch these to using non-relative paths, e.g. `from 'types/actions';`? Makes it easier to refactor/move files around

**Comment by jespino** [comment]
Sorry, but the types import using absolute path have implications for the future use of it as part of other projects. Probably we will need to change it in the future, but for now we are using relative path for types import everywhere. My proposal here is to keep doing that, and when we start to include it in other projects (probably in mattermost-mobile) look for the best way to do it.

**Comment by jwilander** [comment]
Is this something specific to flow?

**Comment by jespino** [comment]
Is not flow specific, is more related to how we compile an distribute mattermost-redux, and at the same time, how we distribute our flow code (using the copy approach). Is something that we can solve with the first mattermost-mobile/mattermost-webapp flow integration. Until then, I think is better to use only 1 approach, and the current one is use the relative paths.

**Comment by jwilander** [comment]
I'm not sure I understand, we use absolute paths relative to the project root for almost all the imports in mattermost-redux. If we're going for consistency then that is what we should do.

**Comment by jespino** [comment]
Consistency with the rest of the flow imports. Let my try to integrate flow in mattermost-mobile to revisit the problem by myself.

**Comment by pepf** [comment]
ok, so we might be able to fix the "flow issue" using [this suggestion](https://github.com/facebook/flow/issues/4186#issuecomment-347078321), but what shall I do with the current code? Just change it to relative for now?..

**Comment by jespino** [comment]
For now, lets back to the relative path, merge it, and then decide what to do with the relative/absolute path problem.

**Comment by pepf** [comment]
done 👌

**Comment by jespino** [comment]
Thanks :sparkles:

**Pull request merged**